src/server/snapshot.cc

```diff
@@ -330,12 +330,14 @@ void SliceSnapshot::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req)
 // no database switch can be performed between those two calls, because they are part of one
 // transaction.
 void SliceSnapshot::OnJournalEntry(const journal::JournalItem& item, bool await) {
-  // We ignore EXEC and NOOP entries because we they have no meaning during
+  // We ignore EXEC entries because we they have no meaning during
   // the LOAD phase on replica.
-  if (item.opcode == journal::Op::NOOP || item.opcode == journal::Op::EXEC)
+  if (item.opcode == journal::Op::EXEC)
     return;
 
-  serializer_->WriteJournalEntry(item.data);
+  if (item.opcode != journal::Op::NOOP) {
```

Comment:
yes, maybe we can comment here @adiholden  ?

Does it mean we pass NOOP with await=true and the fix will allow the flow to reach ` serializer_->WriteJournalEntry(item.data);` ?

Reply:
yes

Reply:
so maybe add a comment here that we allow `TriggerJournalWriteToSink` to flush the serialised chunk because it calls this function with `(NOOP, await=true)`